File: python-sdks/respan-exporter-anthropic-agents/examples/gateway_integration.py
"""
Gateway integration example for respan-exporter-anthropic-agents.

Routes Claude Agent SDK traffic through the Respan gateway instead of
hitting Anthropic directly. This means you only need a Respan API key —
no separate Anthropic key required.

Prerequisites:
  pip install claude-agent-sdk respan-sdk respan-exporter-anthropic-agents

Environment variables (required):
  RESPAN_API_KEY   (or KEYWORDSAI_API_KEY) - your Respan key (used for both
                                              gateway auth and trace ingest)

Environment variables (optional):
  RESPAN_BASE_URL  - gateway base URL (default: https://api.respan.ai)

Run:
  python examples/gateway_integration.py
"""

import asyncio
import os
import sys

from claude_agent_sdk import (
    ClaudeAgentOptions,
    ResultMessage,
    SystemMessage,
    query,
)

from respan_exporter_anthropic_agents import RespanAnthropicAgentsExporter
from respan_exporter_anthropic_agents.utils import (
    extract_session_id_from_system_message,
)


async def main() -> None:
    respan_api_key = os.getenv("RESPAN_API_KEY") or os.getenv("KEYWORDSAI_API_KEY")
    if not respan_api_key:
        print("ERROR: Set RESPAN_API_KEY (or KEYWORDSAI_API_KEY)")
        sys.exit(1)

    base_url = (
        os.getenv("RESPAN_BASE_URL")
        or os.getenv("KEYWORDSAI_BASE_URL")
        or "https://api.respan.ai"
    ).rstrip("/")

    print(f"Gateway base URL: {base_url}")
    print(f"API key: {respan_api_key[:8]}...")

    # Exporter sends traces to the ingest endpoint
    exporter = RespanAnthropicAgentsExporter(
        api_key=respan_api_key,
        base_url=base_url,
    )

    # Route Claude SDK through the Respan gateway
    options = ClaudeAgentOptions(
        permission_mode="bypassPermissions",
        max_turns=1,
        env={
            "ANTHROPIC_BASE_URL": base_url,
            "ANTHROPIC_AUTH_TOKEN": respan_api_key,
            "ANTHROPIC_API_KEY": respan_api_key,
        },
    )
    options = exporter.with_options(options=options)

    print("\nSending query through gateway...\n")

    session_id = None
    result = None
    async for message in query(
        prompt="Reply with exactly: gateway_ok",
        options=options,
    ):
        if isinstance(message, SystemMessage):
            session_id = extract_session_id_from_system_message(
                system_message=message
            )
        if isinstance(message, ResultMessage):
            session_id = message.session_id
            result = message

        await exporter.track_message(message=message, session_id=session_id)
        print(f"  {type(message).__name__}")

    if result:
        print(f"\nResult: subtype={result.subtype}, turns={result.num_turns}")
        print(f"Usage: {result.usage}")

    print(f"\nTraces exported to Respan under session {session_id}")
    print(f"View at: https://platform.respan.ai/traces/{session_id}")


if __name__ == "__main__":
    asyncio.run(main())

File: javascript-sdks/respan-exporter-anthropic-agents/examples/gateway-integration.ts
/**
 * Gateway integration example for @respan/exporter-anthropic-agents.
 *
 * Routes Claude Agent SDK traffic through the Respan gateway instead of
 * hitting Anthropic directly. Only a Respan API key is needed.
 *
 * Prerequisites:
 *   yarn add @anthropic-ai/claude-agent-sdk @respan/respan-sdk @respan/exporter-anthropic-agents
 *
 * Environment variables (required):
 *   RESPAN_API_KEY   (or KEYWORDSAI_API_KEY) - your Respan key
 *
 * Environment variables (optional):
 *   RESPAN_BASE_URL  - gateway base URL (default: https://api.respan.ai)
 *
 * Run:
 *   npx tsx examples/gateway-integration.ts
 */

import { query } from "@anthropic-ai/claude-agent-sdk";
import { RespanAnthropicAgentsExporter } from "../src/index.js";

async function main(): Promise<void> {
  const apiKey = process.env.RESPAN_API_KEY || process.env.KEYWORDSAI_API_KEY;
  if (!apiKey) {
    console.error("ERROR: Set RESPAN_API_KEY (or KEYWORDSAI_API_KEY)");
    process.exit(1);
  }

  const baseUrl = (
    process.env.RESPAN_BASE_URL ||
    process.env.KEYWORDSAI_BASE_URL ||
    "https://api.respan.ai"
  ).replace(/\/+$/, "");

  console.log(`Gateway base URL: ${baseUrl}`);
  console.log(`API key: ${apiKey.slice(0, 8)}...`);

  const exporter = new RespanAnthropicAgentsExporter({
    apiKey,
    endpoint: `${baseUrl}/api/v1/traces/ingest`,
  });

  // Route Claude SDK through the Respan gateway
  const options = exporter.withOptions({
    permissionMode: "bypassPermissions",
    maxTurns: 1,
    env: {
      ANTHROPIC_BASE_URL: baseUrl,
      ANTHROPIC_AUTH_TOKEN: apiKey,
      ANTHROPIC_API_KEY: apiKey,
    },
  } as any);

  console.log("\nSending query through gateway...\n");

  let sessionId: string | undefined;

  for await (const message of query({ prompt: "Reply with exactly: gateway_ok", options })) {
    const msg = message as Record<string, unknown>;

    if (msg.type === "system") {
      const data = (msg.data ?? {}) as Record<string, unknown>;
      sessionId = (data.session_id ?? data.sessionId ?? sessionId) as string;
    }
    if (msg.type === "result") {
      sessionId = (msg.session_id ?? sessionId) as string;
      const usage = msg.usage as Record<string, unknown> | undefined;
      console.log(`  Result: subtype=${msg.subtype}, turns=${msg.num_turns}`);
      if (usage) {
        console.log(`  Usage: input_tokens=${usage.input_tokens}, output_tokens=${usage.output_tokens}`);
      }
    }

    await exporter.trackMessage({ message, sessionId });
    console.log(`  ${msg.type}`);
  }

  console.log(`\nTraces exported to Respan under session ${sessionId}`);
  console.log(`View at: https://platform.respan.ai/traces/${sessionId}`);
}

main().catch(console.error);
